Libraries/Components/ViewPager/ViewPagerAndroid.android.js

```diff
@@ -151,6 +152,13 @@ var ViewPagerAndroid = React.createClass({
   setPage: function(selectedPage) {
     this.setState({
       selectedPage,
+      animatePageTransition: true,
```

Comment:
> This has a hidden side effect of setting animatePageTransition for all future updates of selectedPage.

It actually doesn't have any impact because this only happens for JS-driven page updates, and those always go through `setPage` or `setPageWithoutAnimation` so we set the animated transition prop each time. It's a little bit of an awkward API though I agree. It leads to the `onAfterUpdateTransaction` method above as well because we need to be sure that all of the prop updates for the batch have been applied before actually setting the page on the ViewPager.